### PR TITLE
[2198] Do not blindly set #threads to requested #threads (4-2-stable)

### PR DESCRIPTION
**Pull request by JustinKyleJames**
Test run of the main suite of tests passed.

### CODE REVIEW

File: packaging/resource_suite_s3_nocache.py
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2382]: do we want to add a 0 and 1 to this array?

Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: done

**Comment by trel** [comment]
if everything passes, please squash, no #.   and we'll wait for another set of eyes.

**Comment by JustinKyleJames** [comment]
> if everything passes, please squash, no #. and we'll wait for another set of eyes.

I added an additional test case for a recursive put with -N flag and checksum as that was initially reported as the issue.  It's a little overkill.

I squashed the one additional commit but kept this latest commit so you can take a look at it.

File: packaging/resource_suite_s3_nocache.py
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2409]: double spaces before %

File: packaging/resource_suite_s3_nocache.py
Comment (trel) [inline packaging/resource_suite_s3_nocache.py:2399]: do we usually use "__issue"... in these names?

Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: Usually.  I made the change.

File: packaging/resource_suite_s3_nocache.py
Comment (JustinKyleJames) [reply packaging/resource_suite_s3_nocache.py]: Removed the extra space.

**Comment by alanking** [comment]
Are we planning to do a 4.2.12.2 release?

**Comment by korydraughn** [comment]
No due to 4.2.12 being EOL.

**Comment by JustinKyleJames** [comment]
> Are we planning to do a 4.2.12.2 release?

I think this is important enough to have a release in 4.2.12.

**Comment by trel** [comment]
Agreed.

**Comment by alanking** [comment]
Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

**Comment by JustinKyleJames** [comment]
> Sounds good. I've left a review on the 4-3-stable version. Please let us know if there are any significant differences between the two / whether we should treat this as a distinct review (besides the usual checks).

I will make those changes on this one.  Unfortunately it is a manual process due to the python->python3 update.

I am going to hold off on making the changes for #2204 until that other PR gets approved and merged.

**Comment by JustinKyleJames** [comment]
I made and tested the changes from the 4-3-stable PR to this one - issue #2204 changes excluded for now.

**Comment by korydraughn** [comment]
What's the current status of this PR?

**Comment by JustinKyleJames** [comment]
This pull request is up-to-date.  It is a cherry-pick with necessary alterations for 4.2 - mostly just Python3 -> Python2.

Tests passed.

**Review by alanking — APPROVED** [review]
Compared with https://github.com/irods/irods_resource_plugin_s3/pull/2200 and everything seemed correct to me. # it

**Review by korydraughn — APPROVED** [review]

**Comment by JustinKyleJames** [comment]
> Compared with #2200 and everything seemed correct to me. # it

Done

**Pull request merged**
